src/hotspot/share/opto/macro.cpp

```diff
@@ -274,7 +274,7 @@ static Node *scan_mem_chain(Node *mem, int alias_idx, int offset, Node *start_me
 
 // Generate loads from source of the arraycopy for fields of
 // destination needed at a deoptimization point
-Node* PhaseMacroExpand::make_arraycopy_load(ArrayCopyNode* ac, intptr_t offset, Node* ctl, Node* mem, BasicType ft, const Type *ftype, AllocateNode *alloc) {
+Node* PhaseMacroExpand::make_arraycopy_load(Compile* comp, PhaseIterGVN* igvn, ArrayCopyNode* ac, intptr_t offset, Node* ctl, Node* mem, BasicType ft, const Type *ftype, AllocateNode *alloc) {
```

Comment:
What's the reasoning for passing igvn in instead of using the instance variable? Is there a different PhaseIterGVN needed from call to call?

Reply:
The `value_from_mem_phi` method was made static so that it can be used without an instance of PhaseMacroExpand. Since the method is static we can't use an instance variable inside it.